### PR TITLE
Enable Db parameter support for Structured SP parameters

### LINKED ISSUE #8: Support for Structured parameters
I am trying to pass in a DataTable as a parameter to a stored procedure, and it's not working. I am getting a NotSupportedException.

After looking quickly, it appears to be because this is not supporting SQL Server specifically, which allows for a DataTable to be passed as a parameter with the SqlDbType.Structured.

Is there plan for future support of SQL Server specifically?

### PR DESCRIPTION
**Pull request by korifeich**
Add AddParam method to IStoredProcedureBuilder that accepts DBParameter as entry. It is a possible option to allow to pass Structured parameters in.

resolve #8

### CODE REVIEW

File: StoredProcedureEFCore/StoredProcBuilder.cs [hunk 1]
```diff
@@ -70,7 +81,7 @@ public IStoredProcBuilder ReturnValue<T>(out IOutParam<T> retParam, ParamExtra e
       retParam = AddOutputParamInner(_retParamName, default(T), ParameterDirection.ReturnValue, extra);
       return this;
     }
-
+
```
Comment (verdie-g) [inline StoredProcedureEFCore/StoredProcBuilder.cs]: Trailing spaces

File: StoredProcedureEFCore/IStoredProcBuilder.cs [hunk 2]
```diff
@@ -58,13 +58,20 @@ public interface IStoredProcBuilder : IDisposable
...
+    IStoredProcBuilder AddParam(DbParameter parameter);
+
     /// <summary>
     /// Add return value parameter
     /// </summary>
     /// <typeparam name="T">Type of the parameter. Can be nullable</typeparam>
     /// <param name="retParam">Created parameter. Value will be populated after calling <see cref="Exec(Action{DbDataReader})"/></param>
     /// <returns></returns>
-    IStoredProcBuilder ReturnValue<T>(out IOutParam<T> retParam);
+        IStoredProcBuilder ReturnValue<T>(out IOutParam<T> retParam);
```
Comment (verdie-g) [inline StoredProcedureEFCore/IStoredProcBuilder.cs]: Remove the added spaces

File: StoredProcedureEFCore/StoredProcBuilder.cs [hunk 3]
```diff
@@ -59,6 +59,17 @@ public IStoredProcBuilder AddParam<T>(string name, T val, out IOutParam<T> outPa
       return this;
     }
 
+    public IStoredProcBuilder AddParam(DbParameter parameter)
+    {
+      if(parameter == null)
```
Comment (verdie-g) [inline StoredProcedureEFCore/StoredProcBuilder.cs]: add space before '('

**Comment by verdie-g** [comment]
Wait... to create a DBParameter you need a DbCommand which is not exposed by the IStoredProcBuilder

**Comment by korifeich** [comment]
You can create any concrete realizations of DBParameter like: SqlParameter without DBCommand.

**Pull request merged**